server/src/main/java/org/folio/reservoir/server/UploadService.java

```diff
@@ -24,19 +29,67 @@
 
 public class UploadService {
 
+  private static final String LOCAL_ID = "localId";
   private static final Logger log = LogManager.getLogger(UploadService.class);
 
-  private Future<Void> uploadPayloadStream(ReadStream<JsonObject> upload,
-      IngestWriteStream ingestWriteStream) {
+  private Future<Void> uploadPayloadStream(Vertx vertx, ReadStream<JsonObject> upload,
+      IngestWriteStream ingestWriteStream, ModuleJsonPath jsonPath) {
     Promise<Void> promise = Promise.promise();
     if (ingestWriteStream != null) {
       ingestWriteStream.drainHandler(x -> upload.resume());
     }
     AtomicInteger number = new AtomicInteger();
     AtomicInteger errors = new AtomicInteger();
     upload.exceptionHandler(promise::tryFail);
-    upload.handler(r -> {
-      String localId = r.getString("localId");
+    upload.handler(r ->
+        validateIngestRecord(vertx, jsonPath, r, number, errors) 
+          .compose(rec -> {
+            if (rec == null || ingestWriteStream == null) {
+              return Future.succeededFuture(null);
```

Comment:
```suggestion
              return Future.succeededFuture();
```

Reply:
Yes, this is annoying. I'll check if this can be configured.

Reply:
grep trim .vscode/settings.json 
     "files.trimTrailingWhitespace": true